consensus/hotstuff/cruisectl/block_time_controller_test.go

```diff
@@ -223,59 +216,89 @@ func (bs *BlockTimeControllerSuite) TestInit_EpochSetupPhase() {
-	bs.ctl.proportionalErr.AddObservation(20.0)
-	bs.ctl.integralErr.AddObservation(20.0)
-	err := bs.ctl.measureViewDuration(makeTimedBlock(bs.initialView+1, unittest.IdentifierFixture(), time.Now()))
-	require.NoError(bs.T(), err)
-	assert.NotEqual(bs.T(), bs.config.FallbackProposalDelay, bs.ctl.getProposalTiming())
+	extension := flow.EpochExtension{
+		FirstView: setupFixture.FirstView,
+		FinalView: setupFixture.FinalView,
+	}
+
+	epoch := inmem.NewSetupEpoch(setupFixture, []flow.EpochExtension{extension})
+	bs.epochs.Add(epoch)
+	bs.epochs.Transition()
 
-	// send the event
-	bs.ctl.EpochFallbackModeTriggered(0, nil)
-	// async: should revert to default GetProposalTiming
+	header := unittest.BlockHeaderFixture()
+	bs.state.On("AtHeight", header.Height).Return(&bs.snapshot).Once()
+
+	bs.ctl.EpochExtended(bs.epochCounter, header, extension)
+
+	// expect epoch fallback triggered to be set to true when first extension is encountered indicating we are in epoch fallback mode
 	require.Eventually(bs.T(), func() bool {
-		now := time.Now().UTC()
-		return now.Add(bs.config.FallbackProposalDelay.Load()) == bs.ctl.getProposalTiming().TargetPublicationTime(7, now, unittest.IdentifierFixture())
+		return len(bs.ctl.epochEvents) == 0
 	}, time.Second, time.Millisecond)
 
-	// additional EpochFallbackModeTriggered events should be no-ops
-	// (send capacity+1 events to guarantee one is processed)
-	for i := 0; i <= cap(bs.ctl.epochFallbacks); i++ {
-		bs.ctl.EpochFallbackModeTriggered(0, nil)
+	extensionTargetTime, err := bs.snapshot.Epochs().Current().TargetEndTime()
+	require.NoError(bs.T(), err)
+	extensionTargetDuration, err := bs.snapshot.Epochs().Current().TargetDuration()
+	require.NoError(bs.T(), err)
+	extensionFinalView, err := bs.snapshot.Epochs().Current().FinalView()
+	require.NoError(bs.T(), err)
+
+	assert.Equal(bs.T(), extensionTargetTime, bs.ctl.curEpochTargetEndTime)
+	assert.Equal(bs.T(), extensionTargetDuration, bs.ctl.curEpochTargetDuration)
+	assert.Equal(bs.T(), extensionFinalView, bs.ctl.curEpochFinalView)
+
+	// duplicate events should be no-ops
+	for i := 0; i <= cap(bs.ctl.epochEvents); i++ {
+		bs.ctl.EpochExtended(bs.epochCounter, header, extension)
 	}
-	// state should be unchanged
-	now := time.Now().UTC()
-	assert.Equal(bs.T(), now.Add(bs.config.FallbackProposalDelay.Load()), bs.ctl.getProposalTiming().TargetPublicationTime(12, now, unittest.IdentifierFixture()))
+	require.Eventually(bs.T(), func() bool {
+		return len(bs.ctl.epochEvents) == 0
+	}, time.Second, time.Millisecond)
 
-	// additional OnBlockIncorporated events should be no-ops
-	for i := 0; i <= cap(bs.ctl.incorporatedBlocks); i++ {
-		header := unittest.BlockHeaderFixture(unittest.HeaderWithView(bs.initialView + 1 + uint64(i)))
-		header.ParentID = unittest.IdentifierFixture()
-		bs.ctl.OnBlockIncorporated(model.BlockFromFlow(header))
+	assert.Equal(bs.T(), extensionTargetTime, bs.ctl.curEpochTargetEndTime)
```

Comment:
can we also check the duration here? Just for good measure, so we have all possibly variable quantities checked here:
```suggestion
	assert.Equal(bs.T(), extensionTargetTime, bs.ctl.curEpochTargetEndTime)
	assert.Equal(bs.T(), extensionTargetDuration, bs.ctl.curEpochTargetDuration)
```